src/components/SignUp/SignUpScreen.js

```diff
@@ -1,65 +1,282 @@
+    color: '#fff',
+    '&:hover': {
+      backgroundColor: '#373737',
+    },
+    marginTop: '10%',
+  },
+  greenButton: {
+    backgroundColor: '#53AA48',
+    '&:hover': {
+      backgroundColor: '#53AA48',
+    },
+    marginTop: '10%',
+  },
+  submitButton: {
+    backgroundColor: '#53AA48',
+    '&:hover': {
+      backgroundColor: '#53AA48',
+    },
+    width: '50%',
+  },
+  textBox: {
+    width: '70%',
+  },
+  card: {
+    margin: 16,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    background: 'FFFFFF',
+    borderRadius: '15px',
+    boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',
+  },
+  stepper: {
+    margin: 'auto',
+    width: '60%',
+  },
+  icon: {
+    color: 'red !important',
+  },
+});
+const BUTTONS = [
+  {
+    id: 0,
+    title: 'Buyer',
+    image: Fruit3,
+    fruit: {
+      marginTop: '10%',
+
+      height: '134px',
+      width: '98px',
+      left: '89px',
+      top: '46px',
+    },
+  },
+  {
+    id: 1,
+    title: 'Non-profit or Agency',
+    image: Fruit2,
+
+    fruit: {
+      marginTop: '10%',
+      height: '130px',
+      width: '105px',
+      left: '96px',
+      top: '50px',
+    },
+  },
+  {
+    id: 2,
+    title: 'Seller',
+    image: Fruit1,
+
+    fruit: {
+      marginTop: '10%',
+      height: '141px',
+      width: '91px',
+      left: '72px',
+      top: '39px',
+      // border-radius: '0px',
+    },
+  },
+];
 
 const INITIAL_FORM_STATE = {
-  username: '',
-  fullname: '',
+  firstName: '',
+  lastName: '',
+  email: '',
   password: '',
   confirmPassword: '',
+  org: '',
+  phone: '',
+  zipcode: '',
+  comments: '',
 };
 
+function getSteps() {
+  return ['Step 1', 'Step 2', 'Step 3', 'Confirmation'];
+}
+
+// TODO: Implement popup so when the user presses 'next' at step 3, there is a confirmation message
 export default function SignUpScreen() {
   const [formState, setFormState] = useState(INITIAL_FORM_STATE);
-  const [role, setRole] = React.useState('');
+  // Role mappings: 0 -> Buyer, 1 -> Agency, 2 -> Seller
+  const [role, setRole] = useState(-1);
   const [loading, setLoading] = useState(false);
   const [errorMsg, setErrorMsg] = useState('');
+  const [currentStep, setCurrentStep] = useState(1);
 
-  const roleChange = (event) => {
-    setRole(event.target.value);
-  };
-  const onSubmit = useCallback((event) => {
+  const classes = useStyles();
+
+  const handleChange = useCallback((event) => {
+    event.preventDefault();
+    setFormState(
+      {
+        ...formState,
+        [event.currentTarget.name]: event.currentTarget.value,
+      },
+    );
+  }, [formState, setFormState]);
+
+  const handleSubmit = useCallback((event) => {
     event.preventDefault();
     setLoading(true);
+    let res = null;
     try {
-      /* eslint-disable */
-      const res = signupUser(formState.username, formState.password, formState.fullname, role);
-
+      res = signupUser(formState.email, formState.password,
```

Comment:
I just realized that for buyers, they only have to go through one authentication process. Can you do a check to see if the role is buyer, and if so, make account approval default to "unapproved" and registration approval to default as 
"approved"? That way F2P only has to approve them once.